cloud/cloud_storage_provider.cc

```diff
@@ -19,6 +19,10 @@
 #include "util/stderr_logger.h"
 #include "util/string_util.h"
 
+#ifdef _MSC_VER
+#undef GetObject
```

Comment:
Should this be put into cloud_storage_provider.h and based on OS_WIN?  It seems like everywhere that uses this method might need to do this...

Reply:
I think the best would be to define a port/win32/undefine.h that we would include in every cc. If you put the #undef in a .h, it may be overruled by a later <windows.h> include, or worse, propagate the undef where you don't want it to be propagated.

Reply:
Note that this problem doesn't exist in RocksDB proper as there is no method named GetObject.

Reply:
Since this is a pretty new method (and not likely to be used outside of internal code), would it make sense to rename it and, if so, to what?

Reply:
I would vote to rename CloudStorageProvider.GetObject and CloudStorageProvider.PutObject to CloudStorageProvider.GetCloudObject() and CloudStorageProvider.PutCloudObject() is that sonds ok to you?

Reply:
That sounds like a plan.

Reply:
The "GetObject" method was renamed (to "GetCloudObject").  If you merge up from mainline, you should no longer need the undef.

Can you please merge to latest master and update the PR?

Reply:
Yes I will, thanks!